src/Compiler/Optimize/LowerComputedCollections.fs

```diff
@@ -324,21 +333,21 @@ module Array =
             else
                 expr
 
-        let ilTy, ilBasicTy =
-            let ty = stripMeasuresFromTy g overallElemTy
-
-            if typeEquiv g ty g.int32_ty then g.ilg.typ_Int32, DT_I4
-            elif typeEquiv g ty g.int64_ty then g.ilg.typ_Int64, DT_I8
-            elif typeEquiv g ty g.uint64_ty then g.ilg.typ_UInt64, DT_U8
-            elif typeEquiv g ty g.uint32_ty then g.ilg.typ_UInt32, DT_U4
-            elif typeEquiv g ty g.nativeint_ty then g.ilg.typ_IntPtr, DT_I
-            elif typeEquiv g ty g.unativeint_ty then g.ilg.typ_UIntPtr, DT_U
-            elif typeEquiv g ty g.int16_ty then g.ilg.typ_Int16, DT_I2
-            elif typeEquiv g ty g.uint16_ty then g.ilg.typ_UInt16, DT_U2
-            elif typeEquiv g ty g.sbyte_ty then g.ilg.typ_SByte, DT_I1
-            elif typeEquiv g ty g.byte_ty then g.ilg.typ_Byte, DT_U1
-            elif typeEquiv g ty g.char_ty then g.ilg.typ_Char, DT_U2
-            else error (InternalError ($"Unable to find IL type for integral type '{overallElemTy}'.", m))
+        let stelem =
+            if ilTy = g.ilg.typ_Int32 then I_stelem DT_I4
+            elif ilTy = g.ilg.typ_Int64 then I_stelem DT_I8
+            elif ilTy = g.ilg.typ_UInt64 then I_stelem DT_U8
+            elif ilTy = g.ilg.typ_UInt32 then I_stelem DT_U4
+            elif ilTy = g.ilg.typ_IntPtr then I_stelem DT_I
+            elif ilTy = g.ilg.typ_UIntPtr then I_stelem DT_U
+            elif ilTy = g.ilg.typ_Int16 then I_stelem DT_I2
+            elif ilTy = g.ilg.typ_UInt16 then I_stelem DT_U2
+            elif ilTy = g.ilg.typ_SByte then I_stelem DT_I1
+            elif ilTy = g.ilg.typ_Byte then I_stelem DT_U1
+            elif ilTy = g.ilg.typ_Char then I_stelem DT_U2
+            elif ilTy = g.ilg.typ_Double then I_stelem DT_R8
+            elif ilTy = g.ilg.typ_Single then I_stelem DT_R4
+            else I_stelem_any (ILArrayShape.SingleDimensional, ilTy)
```

Comment:
Why are we replacing an internal error with `I_stelem_any`, is the scenario in which this occurs tested anyhow?

Reply:
Because this now supports any type as the return type of the mapping (and thus the type parameter for the resulting array).

E.g.,

```fsharp
[|for n in 1..10 -> n, n * n|]      // (int * int) array
[|for n in 1..10 -> box n|]         // obj array
[|for n in 1..10 -> {| N = n |}|]   // {| N : int |} array
```

Reply:
Thanks, I see now.